### ISSUE TITLE
Replace deprecated CUDA function in GPU library

### ISSUE THREAD

**Issue by akohlmey** [lammps/lammps#1035]
## Summary

When compiling the GPU package library with CUDA 9.x, the following type of warning appears:
```
lal_preprocessor.h(129): warning: function "__shfl_xor(int, int, int)"
/opt/tools/cuda-9.0.176/include/sm_30_intrinsics.hpp(221): here was declared deprecated ("__shfl_xor() is deprecated in favor of __shfl_xor_sync() and may be removed in a future release (Use -Wno-deprecated-declarations to suppress this warning).")

lal_preprocessor.h(130): warning: function "__shfl_xor(int, int, int)"
/opt/tools/cuda-9.0.176/include/sm_30_intrinsics.hpp(221): here was declared deprecated ("__shfl_xor() is deprecated in favor of __shfl_xor_sync() and may be removed in a future release (Use -Wno-deprecated-declarations to suppress this warning).")
```

We should update this part of the code accordingly to be future proof. Also, it needs to be checked what is the minimum CUDA toolkit version required, that provides this feature. KOKKOS currently requires a minimum of CUDA toolkit v7.5. Requiring a newer toolkit version will also mean, that support for older hardware will be dropped. AFAIK, already the latest CUDA drivers drop support for Fermi and older.

## Type of Issue

Suggestion for an Enhancement

**Comment by rbberger** [comment]
already got that one fixed, now looking into other deprecated functions as well

**Comment by akohlmey** [comment]
fixed in #1046
